<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Constructing marker genes from purified scRNA-seq data • cellassign</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Constructing marker genes from purified scRNA-seq data">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cellassign</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/constructing-markers-from-purified-data.html">Constructing marker genes from purified scRNA-seq data</a>
    </li>
    <li>
      <a href="../articles/introduction-to-cellassign.html">Assigning single-cells to known cell types with CellAssign</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Constructing marker genes from purified scRNA-seq data</h1>
                        <h4 class="author">Allen W Zhang and Kieran R Campbell</h4>
            
            <h4 class="date">April 2019</h4>
      
      
      <div class="hidden name"><code>constructing-markers-from-purified-data.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>In many situations, marker genes for cell types are either known <em>a priori</em> as expert knowledge, or can be curated through databases such as the <a href="http://biocc.hrbmu.edu.cn/CellMarker/">Cellmark</a> database. Alternatively, if purified expression data exists (either in bulk or single-cell form), it is possible to quickly derive marker genes using the <code>findMarkers</code> function in the <a href="http://bioconductor.org/packages/release/bioc/html/scran.html">scran</a> R package.</p>
<p>Below we detail a case study in deriving marker genes through a differential expression approach.</p>
</div>
<div id="data" class="section level1">
<h1 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h1>
<div id="overview-1" class="section level2">
<h2 class="hasAnchor">
<a href="#overview-1" class="anchor"></a>Overview</h2>
<p>We take bulk RNA-seq data from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5389713/">Holik et al. Nucleic Acids Research 2017</a> to derive marker genes for 3 different cell lines. This is packaged with <code>cellassign</code> as <code>holik_data</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(holik_data)</a></code></pre></div>
<p>which contains a matrix of counts, where each row is a gene (index by entrez ID) and each column is a sample:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(holik_data<span class="op">$</span>counts[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>])</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#&gt;    RSCE_10_BC2CTUACXX_AGTTCC_L006_R1.bam RSCE_12_BC2CTUACXX_TGACCA_L005_R1.bam</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; 1                                     13                                    12</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; 2                                      0                                     0</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; 9                                    346                                   286</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; 10                                     0                                     1</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; 12                                    10                                    17</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt; 13                                     1                                     0</span></a></code></pre></div>
<p>as well as a vector with the cell line of origin for each sample:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(holik_data<span class="op">$</span>cell_line)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; [1] "HCC827" "HCC827" "H2228"  "H2228"  "H838"   "H838"</span></a></code></pre></div>
</div>
<div id="preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#preparation" class="anchor"></a>Preparation</h2>
<p>We first provide a map from entrez IDs to gene symbols:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">entrez_map &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, </a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                     <span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(holik_data<span class="op">$</span>counts)), </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                     <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"SYMBOL"</span>), <span class="st">"ENTREZID"</span>)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">gene_annotations &lt;-<span class="st"> </span>entrez_map <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">rename</a></span>(<span class="dt">GeneID=</span>ENTREZID,</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">                <span class="dt">Symbol=</span>SYMBOL)</a></code></pre></div>
<p>Then construct the <code>DGEList</code> object for input to <code>limma voom</code>, filtering out lowly expressed genes:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">dge &lt;-<span class="st"> </span><span class="kw">DGEList</span>(<span class="dt">counts =</span> holik_data<span class="op">$</span>counts, </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">               <span class="dt">group =</span> holik_data<span class="op">$</span>cell_line, </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">               <span class="dt">genes =</span> gene_annotations, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">               <span class="dt">remove.zeros =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; Removing 3620 rows with all zero counts</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">genes_to_keep &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="kw">cpm</span>(dge<span class="op">$</span>counts) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">dge_filt &lt;-<span class="st"> </span>dge[genes_to_keep,]</a></code></pre></div>
<p>and finally calculate the normalization factors:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">dge_filt &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(dge_filt, <span class="dt">method=</span><span class="st">"TMM"</span>)</a></code></pre></div>
</div>
</div>
<div id="differential-expression" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-expression" class="anchor"></a>Differential expression</h1>
<p>We next perform differential expression using Limma Voom on a subset of 3 samples: HCC827, H2228, H1975:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">dge_subset &lt;-<span class="st"> </span>dge_filt[,dge_filt<span class="op">$</span>samples<span class="op">$</span>group <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"HCC827"</span>, <span class="st">"H2228"</span>, <span class="st">"H1975"</span>)]</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">design &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(<span class="op">~</span><span class="st"> </span><span class="dv">0</span><span class="op">+</span>dge_subset<span class="op">$</span>samples<span class="op">$</span>group)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(design) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/levels.html">levels</a></span>(dge_subset<span class="op">$</span>samples<span class="op">$</span>group)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">v &lt;-<span class="st"> </span><span class="kw">voom</span>(dge_subset, design)</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(v, design)</a></code></pre></div>
<p>Next, fit constrasts to find differentially expressed genes between cell types:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">contrast.matrix &lt;-<span class="st"> </span><span class="kw">makeContrasts</span>(H2228 <span class="op">-</span><span class="st"> </span>H1975, </a>
<a class="sourceLine" id="cb8-2" data-line-number="2">                                 HCC827 <span class="op">-</span><span class="st"> </span>H1975, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                                 HCC827 <span class="op">-</span><span class="st"> </span>H2228, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                                 <span class="dt">levels =</span> design)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">fit2 &lt;-<span class="st"> </span><span class="kw">contrasts.fit</span>(fit, contrast.matrix)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">fit2 &lt;-<span class="st"> </span><span class="kw">eBayes</span>(fit2)</a></code></pre></div>
<p>Finally, compute gene summary statistics and filter to only significantly differentially expressed geens (FDR &lt; 0.05):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">tt &lt;-<span class="st"> </span><span class="kw">topTable</span>(fit2, <span class="dt">n=</span><span class="ot">Inf</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">tt_sig &lt;-<span class="st"> </span>tt <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(adj.P.Val <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(tt_sig)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt;   GeneID    Symbol H2228...H1975 HCC827...H1975 HCC827...H2228  AveExpr</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; 1   1956      EGFR    0.68777340       4.303436       3.615662 9.208711</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; 2  23480    SEC61G   -0.40115418       4.414580       4.815734 7.784072</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; 3  81552     VOPP1   -0.09640406       4.347392       4.443796 6.998559</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; 4   1362       CPD    2.12265278      -2.103571      -4.226224 8.704180</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; 5 729086 LOC729086   -0.04338772       4.384976       4.428363 6.555193</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; 6  55915    LANCL2   -0.57454448       4.021963       4.596508 6.568625</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt;          F      P.Value    adj.P.Val</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt; 1 2026.616 3.447052e-12 2.982899e-08</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; 2 2003.024 3.623981e-12 2.982899e-08</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; 3 1766.548 6.199752e-12 3.232724e-08</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; 4 1671.376 7.854996e-12 3.232724e-08</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; 5 1506.362 1.224645e-11 3.628466e-08</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; 6 1479.497 1.322488e-11 3.628466e-08</span></a></code></pre></div>
</div>
<div id="marker-gene-derivation" class="section level1">
<h1 class="hasAnchor">
<a href="#marker-gene-derivation" class="anchor"></a>Marker gene derivation</h1>
<p>To derive marker genes, we first create a log fold change matrix using H1975 as the baseline expression:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">lfc_table &lt;-<span class="st"> </span>tt_sig[,<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"H2228...H1975"</span>, <span class="st">"HCC827...H1975"</span>)]</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">lfc_table &lt;-<span class="st"> </span>lfc_table <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">H1975=</span><span class="dv">0</span>,</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                <span class="dt">H2228=</span>H2228...H1975,</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">                <span class="dt">HCC827=</span>HCC827...H1975) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(H1975, H2228, HCC827)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(lfc_table) &lt;-<span class="st"> </span>tt_sig<span class="op">$</span>GeneID</a></code></pre></div>
<p>Then, for each gene, we subtract the minimum log fold change, as we care about overexpression of genes relative to some minimum expression level, as this defines a marker gene:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">lfc_table &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(lfc_table)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">lfc_table &lt;-<span class="st"> </span>lfc_table <span class="op">-</span><span class="st"> </span><span class="kw">rowMins</span>(lfc_table)</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">lfc_table &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(lfc_table)</a></code></pre></div>
<p>We now define a helper function for turning log fold changes into a binary matrix. This takes a matrix and a threshold, and any values less than or equal to the threshold are set to 0, and all others to 1:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">binarize &lt;-<span class="st"> </span><span class="cf">function</span>(x, threshold) {</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">  x[x <span class="op">&lt;=</span><span class="st"> </span>threshold] &lt;-<span class="st"> </span><span class="op">-</span><span class="ot">Inf</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3">  x[x <span class="op">&gt;</span><span class="st"> </span><span class="op">-</span><span class="ot">Inf</span>] &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4">  x[x <span class="op">==</span><span class="st"> </span><span class="op">-</span><span class="ot">Inf</span>] &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5">  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span>(x)</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">}</a></code></pre></div>
<p>Next, we implement a basic procedure for binarizing this matrix. Essentially, we look for the largest ‘gap’ in expression for each gene, and the cell types with expression above this gap are designated has having that gene as a marker:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># Find the biggest difference</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">maxdiffs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(lfc_table, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/diff.html">diff</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(x))))</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="co">#</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5">thres_vals &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(lfc_table, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="kw"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(x)[<span class="kw"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/diff.html">diff</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(x)))])</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">expr_mat_thres &lt;-<span class="st"> </span>plyr<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/plyr/man/rbind.fill.html">rbind.fill</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(lfc_table), <span class="cf">function</span>(i) {</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">  <span class="kw">binarize</span>(lfc_table[i,], thres_vals[i])</a>
<a class="sourceLine" id="cb13-8" data-line-number="8">}))</a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(expr_mat_thres) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(lfc_table)</a>
<a class="sourceLine" id="cb13-10" data-line-number="10">marker_gene_mat &lt;-<span class="st"> </span>expr_mat_thres[(maxdiffs <span class="op">&gt;=</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(maxdiffs, <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(.<span class="dv">99</span>))) </a>
<a class="sourceLine" id="cb13-11" data-line-number="11">                                  <span class="op">&amp;</span><span class="st"> </span>(thres_vals <span class="op">&lt;=</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log</a></span>(<span class="dv">2</span>)),] <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-12" data-line-number="12"><span class="st">  </span>as.matrix</a></code></pre></div>
<p>Finally, we add back in gene symbols rather than entrez ids:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span>({</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">  symbols &lt;-<span class="st"> </span>plyr<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/plyr/man/mapvalues.html">mapvalues</a></span>(</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">    <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(marker_gene_mat),</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">    <span class="dt">from =</span> gene_annotations<span class="op">$</span>GeneID,</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">    <span class="dt">to =</span> gene_annotations<span class="op">$</span>Symbol</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">  )</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">})</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"></a>
<a class="sourceLine" id="cb14-9" data-line-number="9">is_na &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(symbols)</a>
<a class="sourceLine" id="cb14-10" data-line-number="10"></a>
<a class="sourceLine" id="cb14-11" data-line-number="11">marker_gene_mat &lt;-<span class="st"> </span>marker_gene_mat[<span class="op">!</span>is_na,]</a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(marker_gene_mat) &lt;-<span class="st"> </span>symbols[<span class="op">!</span>is_na]</a></code></pre></div>
<p>And there we have a marker gene matrix for our cell types:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(marker_gene_mat)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="co">#&gt;          H1975 H2228 HCC827</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="co">#&gt; NRK          0     0      1</span></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="co">#&gt; MTAP         1     0      1</span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="co">#&gt; CP           0     1      0</span></a>
<a class="sourceLine" id="cb15-6" data-line-number="6"><span class="co">#&gt; HIST1H3F     1     0      1</span></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co">#&gt; HIST1H4E     1     0      1</span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="co">#&gt; ADIRF        1     1      0</span></a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">pheatmap</span>(<span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(marker_gene_mat))</a></code></pre></div>
<p><img src="constructing-markers-from-purified-data_files/figure-html/unnamed-chunk-17-1.png" width="960"></p>
<p>Note that the expression data used for input to <code>CellAssign</code> should use only these as input.</p>
</div>
<div id="technical" class="section level1">
<h1 class="hasAnchor">
<a href="#technical" class="anchor"></a>Technical</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#&gt; R version 3.6.0 (2019-04-26)</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#&gt; Running under: macOS  10.15.2</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="co">#&gt; [1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8</span></a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="co">#&gt;  [1] cellassign_0.99.16   pheatmap_1.0.12      matrixStats_0.55.0  </span></a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="co">#&gt;  [4] edgeR_3.26.8         org.Hs.eg.db_3.8.2   AnnotationDbi_1.46.1</span></a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="co">#&gt;  [7] IRanges_2.18.3       S4Vectors_0.22.1     Biobase_2.44.0      </span></a>
<a class="sourceLine" id="cb17-21" data-line-number="21"><span class="co">#&gt; [10] BiocGenerics_0.30.0  limma_3.40.6         magrittr_1.5        </span></a>
<a class="sourceLine" id="cb17-22" data-line-number="22"><span class="co">#&gt; [13] BiocStyle_2.12.0    </span></a>
<a class="sourceLine" id="cb17-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="co">#&gt;  [1] Rcpp_1.0.3                  locfit_1.5-9.1             </span></a>
<a class="sourceLine" id="cb17-26" data-line-number="26"><span class="co">#&gt;  [3] lattice_0.20-38             assertthat_0.2.1           </span></a>
<a class="sourceLine" id="cb17-27" data-line-number="27"><span class="co">#&gt;  [5] zeallot_0.1.0               rprojroot_1.3-2            </span></a>
<a class="sourceLine" id="cb17-28" data-line-number="28"><span class="co">#&gt;  [7] digest_0.6.23               plyr_1.8.5                 </span></a>
<a class="sourceLine" id="cb17-29" data-line-number="29"><span class="co">#&gt;  [9] R6_2.4.1                    GenomeInfoDb_1.20.0        </span></a>
<a class="sourceLine" id="cb17-30" data-line-number="30"><span class="co">#&gt; [11] backports_1.1.5             RSQLite_2.2.0              </span></a>
<a class="sourceLine" id="cb17-31" data-line-number="31"><span class="co">#&gt; [13] evaluate_0.14               pillar_1.4.3               </span></a>
<a class="sourceLine" id="cb17-32" data-line-number="32"><span class="co">#&gt; [15] zlibbioc_1.30.0             tfruns_1.4                 </span></a>
<a class="sourceLine" id="cb17-33" data-line-number="33"><span class="co">#&gt; [17] rlang_0.4.2                 rstudioapi_0.10            </span></a>
<a class="sourceLine" id="cb17-34" data-line-number="34"><span class="co">#&gt; [19] whisker_0.4                 blob_1.2.0                 </span></a>
<a class="sourceLine" id="cb17-35" data-line-number="35"><span class="co">#&gt; [21] Matrix_1.2-18               reticulate_1.13            </span></a>
<a class="sourceLine" id="cb17-36" data-line-number="36"><span class="co">#&gt; [23] rmarkdown_2.0               pkgdown_1.4.1              </span></a>
<a class="sourceLine" id="cb17-37" data-line-number="37"><span class="co">#&gt; [25] desc_1.2.0                  BiocParallel_1.18.1        </span></a>
<a class="sourceLine" id="cb17-38" data-line-number="38"><span class="co">#&gt; [27] stringr_1.4.0               RCurl_1.95-4.13            </span></a>
<a class="sourceLine" id="cb17-39" data-line-number="39"><span class="co">#&gt; [29] bit_1.1-15.1                munsell_0.5.0              </span></a>
<a class="sourceLine" id="cb17-40" data-line-number="40"><span class="co">#&gt; [31] DelayedArray_0.10.0         compiler_3.6.0             </span></a>
<a class="sourceLine" id="cb17-41" data-line-number="41"><span class="co">#&gt; [33] xfun_0.12                   pkgconfig_2.0.3            </span></a>
<a class="sourceLine" id="cb17-42" data-line-number="42"><span class="co">#&gt; [35] base64enc_0.1-3             tensorflow_2.0.0           </span></a>
<a class="sourceLine" id="cb17-43" data-line-number="43"><span class="co">#&gt; [37] htmltools_0.4.0             tidyselect_0.2.5           </span></a>
<a class="sourceLine" id="cb17-44" data-line-number="44"><span class="co">#&gt; [39] SummarizedExperiment_1.14.1 tibble_2.1.3               </span></a>
<a class="sourceLine" id="cb17-45" data-line-number="45"><span class="co">#&gt; [41] GenomeInfoDbData_1.2.1      bookdown_0.17              </span></a>
<a class="sourceLine" id="cb17-46" data-line-number="46"><span class="co">#&gt; [43] dplyr_0.8.3                 crayon_1.3.4               </span></a>
<a class="sourceLine" id="cb17-47" data-line-number="47"><span class="co">#&gt; [45] MASS_7.3-51.5               bitops_1.0-6               </span></a>
<a class="sourceLine" id="cb17-48" data-line-number="48"><span class="co">#&gt; [47] grid_3.6.0                  jsonlite_1.6               </span></a>
<a class="sourceLine" id="cb17-49" data-line-number="49"><span class="co">#&gt; [49] gtable_0.3.0                lifecycle_0.1.0            </span></a>
<a class="sourceLine" id="cb17-50" data-line-number="50"><span class="co">#&gt; [51] DBI_1.1.0                   scales_1.1.0               </span></a>
<a class="sourceLine" id="cb17-51" data-line-number="51"><span class="co">#&gt; [53] stringi_1.4.5               XVector_0.24.0             </span></a>
<a class="sourceLine" id="cb17-52" data-line-number="52"><span class="co">#&gt; [55] fs_1.3.1                    vctrs_0.2.1                </span></a>
<a class="sourceLine" id="cb17-53" data-line-number="53"><span class="co">#&gt; [57] RColorBrewer_1.1-2          tools_3.6.0                </span></a>
<a class="sourceLine" id="cb17-54" data-line-number="54"><span class="co">#&gt; [59] bit64_0.9-7                 glue_1.3.1                 </span></a>
<a class="sourceLine" id="cb17-55" data-line-number="55"><span class="co">#&gt; [61] purrr_0.3.3                 yaml_2.2.0                 </span></a>
<a class="sourceLine" id="cb17-56" data-line-number="56"><span class="co">#&gt; [63] colorspace_1.4-1            BiocManager_1.30.10        </span></a>
<a class="sourceLine" id="cb17-57" data-line-number="57"><span class="co">#&gt; [65] GenomicRanges_1.36.1        memoise_1.1.0              </span></a>
<a class="sourceLine" id="cb17-58" data-line-number="58"><span class="co">#&gt; [67] knitr_1.27</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li>
<a href="#data">Data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#overview-1">Overview</a></li>
      <li><a href="#preparation">Preparation</a></li>
      </ul>
</li>
      <li><a href="#differential-expression">Differential expression</a></li>
      <li><a href="#marker-gene-derivation">Marker gene derivation</a></li>
      <li><a href="#technical">Technical</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Allen Zhang, Kieran Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
