<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Assigning single-cells to known cell types with CellAssign • cellassign</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Assigning single-cells to known cell types with CellAssign">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cellassign</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/constructing-markers-from-purified-data.html">Constructing marker genes from purified scRNA-seq data</a>
    </li>
    <li>
      <a href="../articles/introduction-to-cellassign.html">Assigning single-cells to known cell types with CellAssign</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Assigning single-cells to known cell types with CellAssign</h1>
                        <h4 class="author">Allen W Zhang and Kieran R Campbell</h4>
            
            <h4 class="date">October 2019</h4>
      
      
      <div class="hidden name"><code>introduction-to-cellassign.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p><code>cellassign</code> assigns cells measured using single cell RNA sequencing to known cell types based on marker gene information. Unlike other<br>
methods for assigning cell types from single cell RNA-seq data, <code>cellassign</code> does not require labeled single cell or purified bulk expression data – <code>cellassign</code> only needs to know whether or not each given gene is a marker of each cell type:</p>
<p><img src="cellassign_overview.png" width="668"></p>
<p>Inference is performed using <a href="http://tensorflow.org/">Tensorflow</a>. For more details please see the <a href="https://www.biorxiv.org/content/10.1101/521914v1">manuscript</a>.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p><code>cellassign</code> depends on <code>tensorflow</code>, which can be installed as follows:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"tensorflow"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(tensorflow)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/tensorflow/man/install_tensorflow.html">install_tensorflow</a></span>(<span class="dt">extra_packages =</span> <span class="st">"tensorflow-probability"</span>)</a></code></pre></div>
<p>Please ensure this installs version 2 of tensorflow. You can check this by calling</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">tensorflow<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/tensorflow/man/tf_config.html">tf_config</a></span>()</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#&gt; TensorFlow v2.1.0 (/usr/local/lib/python3.7/site-packages/tensorflow)</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; Python v3.7 (~/.virtualenvs/r-reticulate/bin/python)</span></a></code></pre></div>
<p>You can confirm that the installation succeeded by running:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sess =<span class="st"> </span>tf<span class="op">$</span><span class="kw">Session</span>()</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">hello &lt;-<span class="st"> </span>tf<span class="op">$</span><span class="kw">constant</span>(<span class="st">'Hello, TensorFlow!'</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">sess<span class="op">$</span><span class="kw">run</span>(hello)</a></code></pre></div>
<p>Note that the <code>tf</code> object is created automatically when the <code>tensorflow</code> library is loaded to provide access to the Tensorflow interface.</p>
<p>For more details see the <a href="https://tensorflow.rstudio.com/tensorflow/articles/installation.html">Rstudio page on tensorflow installation</a>.</p>
<p><code>cellassign</code> can then be installed through Bioconductor via</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">'cellassign'</span>)</a></code></pre></div>
<p>or the development version through github using the <code>devtools</code> package :</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span>(<span class="st">"Irrationone/cellassign"</span>)</a></code></pre></div>
</div>
<div id="basic-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-usage" class="anchor"></a>Basic usage</h1>
<p>We begin by illustrating basic usage of <code>cellassign</code> on some example data bundled with the package. First, load the relevant libraries:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(cellassign)</a></code></pre></div>
<p>We use an example <code>SingleCellExperiment</code> consisting of 200 genes and 500 cells:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(example_sce)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(example_sce)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; dim: 200 500 </span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; metadata(20): params params ... params params</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; assays(6): BatchCellMeans BaseCellMeans ... TrueCounts counts</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; rownames(200): Gene1 Gene2 ... Gene199 Gene200</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; rowData names(6): Gene BaseGeneMean ... DEFacGroup1 DEFacGroup2</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; colnames(500): Cell1 Cell2 ... Cell499 Cell500</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="co">#&gt; colData names(5): Cell Batch Group ExpLibSize EM_group</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="co">#&gt; spikeNames(0):</span></a></code></pre></div>
<p>The true cell types are annotated for convenience in the <code>Group</code> slot of the <code>SingleCellExperiment</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(example_sce<span class="op">$</span>Group))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#&gt; [1] "Group1" "Group2" "Group2" "Group1" "Group1" "Group1"</span></a></code></pre></div>
<p>Also provided is an example gene-by-cell-type binary matrix, whose entries are 1 if a gene is a marker for a given cell type and 0 otherwise:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(example_marker_mat)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(example_marker_mat)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#&gt;        Group1 Group2</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#&gt; Gene1       1      0</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#&gt; Gene2       0      1</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#&gt; Gene3       1      0</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; Gene4       1      0</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; Gene5       1      0</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; Gene6       0      1</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; Gene7       0      1</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; Gene8       0      1</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; Gene9       0      1</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; Gene10      1      0</span></a></code></pre></div>
<p>We further require size factors for each cell. These are stored in <code><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SCE-sizeFactors.html">sizeFactors(example_sce)</a></code> - for your data we recommend computing them using the <code>computeSumFactors</code> function from the <code>scran</code> package. Note: <strong>it is highly recommended to compute size factors using the full set of genes, before subsetting to markers for input to cellassign</strong>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">s &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SCE-sizeFactors.html">sizeFactors</a></span>(example_sce)</a></code></pre></div>
<p>We then call <code>cellassign</code> using the <code><a href="https://rdrr.io/pkg/cellassign/man/cellassign.html">cellassign()</a></code> function, passing in the above information. <strong>It is critical that gene expression data containing only marker genes is used as input to cellassign</strong>. We do this here by subsetting the input <code>SingleCellExperiment</code> using the row names (gene names) of the marker matrix. This also ensures that the order of the genes in the gene expression data matches the order of the genes in the marker matrix.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/cellassign.html">cellassign</a></span>(<span class="dt">exprs_obj =</span> example_sce[<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(example_marker_mat),], </a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                  <span class="dt">marker_gene_info =</span> example_marker_mat, </a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                  <span class="dt">s =</span> s, </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                  <span class="dt">learning_rate =</span> <span class="fl">1e-2</span>, </a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                  <span class="dt">shrinkage =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">                  <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>This returns a <code>cellassign</code> object:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(fit)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">#&gt; A cellassign fit for 500 cells, 10 genes, 2 cell types with 0 covariates</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#&gt;             To access cell types, call celltypes(x)</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#&gt;             To access cell type probabilities, call cellprobs(x)</span></a></code></pre></div>
<p>We can access the maximum likelihood estimates (MLE) of cell type using the <code>celltypes</code> function:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/celltypes.html">celltypes</a></span>(fit)))</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co">#&gt; [1] "Group1" "Group2" "Group2" "Group1" "Group1" "Group1"</span></a></code></pre></div>
<p>and all MLE parameters using <code>mleparams</code>:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/mleparams.html">mleparams</a></span>(fit)))</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt; List of 9</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt;  $ delta  : num [1:10, 1:2] 2.32 0 2.5 2.74 2.82 ...</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#&gt;   .. ..$ : chr [1:10] "Gene1" "Gene2" "Gene3" "Gene4" ...</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt;   .. ..$ : chr [1:2] "Group1" "Group2"</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt;  $ beta   : num [1:10, 1] 0.487 -0.261 -1.013 1.328 1.537 ...</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt;   .. ..$ : chr [1:10] "Gene1" "Gene2" "Gene3" "Gene4" ...</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt;   .. ..$ : NULL</span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt;  $ phi    : num [1:500, 1:10, 1:2] 1.81 1.81 1.8 1.8 1.8 ...</span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt;  $ gamma  : num [1:500, 1:2] 1.00 4.50e-147 8.21e-45 1.00 1.00 ...</span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt;   .. ..$ : NULL</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt;   .. ..$ : chr [1:2] "Group1" "Group2"</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt;  $ mu     : num [1:500, 1:10, 1:2] 22.6 80.8 11.5 15.5 15.8 ...</span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt;  $ a      : num [1:10(1d)] 0.989 1.053 1.125 1.209 1.307 ...</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt;  $ theta  : num [1:2(1d)] 0.472 0.528</span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 1</span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#&gt;   .. ..$ : chr [1:2] "Group1" "Group2"</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#&gt;  $ ld_mean: num 1</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="co">#&gt;  $ ld_var : num 0.785</span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="co">#&gt; NULL</span></a></code></pre></div>
<p>We can also visualize the probabilities of assignment using the <code>cellprobs</code> function that returns a probability matrix for each cell and cell type:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">pheatmap<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/cellprobs.html">cellprobs</a></span>(fit))</a></code></pre></div>
<p><img src="introduction-to-cellassign_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>Finally, since this is simulated data we can check the concordance with the true group values:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/table.html">table</a></span>(example_sce<span class="op">$</span>Group, <span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/celltypes.html">celltypes</a></span>(fit)))</a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#&gt;         </span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#&gt;          Group1 Group2</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#&gt;   Group1    236      0</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#&gt;   Group2      0    264</span></a></code></pre></div>
</div>
<div id="example-set-of-markers-for-tumour-microenvironment" class="section level1">
<h1 class="hasAnchor">
<a href="#example-set-of-markers-for-tumour-microenvironment" class="anchor"></a>Example set of markers for tumour microenvironment</h1>
<p>A set of example markers are included with the <code>cellassign</code> package for common cell types in the human tumour microenvironment. Users should be aware that</p>
<ol style="list-style-type: decimal">
<li>This set is provided as an <em>example</em> only and we recommend researchers derive marker gene sets for their own use</li>
<li>The <code>cellassign</code> workflow is typically iterative, including ensuring all markers are expressed in your expression data, and removing cell types from the input marker matrix that do not appear to be present</li>
</ol>
<p>The marker genes are available for the following cell types:</p>
<ul>
<li>B cells</li>
<li>T cells</li>
<li>Cytotoxic T cells</li>
<li>Monocyte/Macrophage</li>
<li>Epithelial cells</li>
<li>Myofibroblasts</li>
<li>Vascular smooth muscle cells</li>
<li>Endothelial cells</li>
</ul>
<p>These can be accessed by calling</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(example_TME_markers)</a></code></pre></div>
<p>Note that this is a list of two marker lists:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(example_TME_markers)</a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#&gt; [1] "symbol"  "ensembl"</span></a></code></pre></div>
<p>Where <code>symbol</code> contains gene symbols:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(example_TME_markers<span class="op">$</span>symbol, <span class="dt">n =</span> <span class="dv">4</span>), head, <span class="dt">n =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#&gt; $`B cells`</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#&gt; [1] "VIM"   "MS4A1" "CD79A" "PTPRC"</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#&gt; $`T cells`</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#&gt; [1] "VIM"  "CD2"  "CD3D" "CD3E"</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">#&gt; $`Cytotoxic T cells`</span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">#&gt; [1] "VIM"  "CD2"  "CD3D" "CD3E"</span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb19-11" data-line-number="11"><span class="co">#&gt; $`Monocyte/Macrophage`</span></a>
<a class="sourceLine" id="cb19-12" data-line-number="12"><span class="co">#&gt; [1] "VIM"    "CD14"   "FCGR3A" "CD33"</span></a></code></pre></div>
<p>and <code>ensembl</code> contains the equivalent ensembl gene ids:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(example_TME_markers<span class="op">$</span>ensembl, <span class="dt">n =</span> <span class="dv">4</span>), head, <span class="dt">n =</span> <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">#&gt; $`B cells`</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="co">#&gt;               VIM             MS4A1             CD79A             PTPRC </span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="co">#&gt; "ENSG00000026025" "ENSG00000156738" "ENSG00000105369" "ENSG00000081237" </span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="co">#&gt; $`T cells`</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="co">#&gt;               VIM               CD2              CD3D              CD3E </span></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="co">#&gt; "ENSG00000026025" "ENSG00000116824" "ENSG00000167286" "ENSG00000198851" </span></a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-10" data-line-number="10"><span class="co">#&gt; $`Cytotoxic T cells`</span></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="co">#&gt;               VIM               CD2              CD3D              CD3E </span></a>
<a class="sourceLine" id="cb20-12" data-line-number="12"><span class="co">#&gt; "ENSG00000026025" "ENSG00000116824" "ENSG00000167286" "ENSG00000198851" </span></a>
<a class="sourceLine" id="cb20-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb20-14" data-line-number="14"><span class="co">#&gt; $`Monocyte/Macrophage`</span></a>
<a class="sourceLine" id="cb20-15" data-line-number="15"><span class="co">#&gt;               VIM              CD14            FCGR3A              CD33 </span></a>
<a class="sourceLine" id="cb20-16" data-line-number="16"><span class="co">#&gt; "ENSG00000026025" "ENSG00000170458" "ENSG00000203747" "ENSG00000105383"</span></a></code></pre></div>
<p>To use these with <code>cellassign</code> we can turn them into the binary marker by cell type matrix:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">marker_mat &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/marker_list_to_mat.html">marker_list_to_mat</a></span>(example_TME_markers<span class="op">$</span>ensembl)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"></a>
<a class="sourceLine" id="cb21-3" data-line-number="3">marker_mat[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#&gt;                 B cells T cells Cytotoxic T cells</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#&gt; ENSG00000010610       0       1                 0</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#&gt; ENSG00000026025       1       1                 1</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">#&gt; ENSG00000039068       0       0                 0</span></a></code></pre></div>
<p><em>Important</em>: the single cell experiment or input gene expression matrix should be subset accordingly to match the rows of the marker input matrix, e.g. if <code>sce</code> is a <code>SingleCellExperiment</code> with ensembl IDs as rownames then call</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">sce_marker &lt;-<span class="st"> </span>sce[<span class="kw"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(marker_mat), <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(sce)),]</a></code></pre></div>
<p>Note that the rows in the single cell experiment or gene expression matrix should be ordered identically to those in the marker input matrix.</p>
<p>You can the proceed using <code>cellassign</code> as before.</p>
</div>
<div id="advanced-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#advanced-usage" class="anchor"></a>Advanced usage</h1>
<div id="options-for-a-cellassign-call" class="section level2">
<h2 class="hasAnchor">
<a href="#options-for-a-cellassign-call" class="anchor"></a>Options for a <code>cellassign()</code> call</h2>
<p>There are several options to a call to <code>cellassign</code> that can alter the results:</p>
<ul>
<li>
<code>min_delta</code>: the minimum log-fold change in expression above which a<br>
genemust be over-expressed in the cells of which it is a marker compared to all others</li>
<li>
<code>X</code>: a covariate matrix, see section below</li>
<li>
<code>shrinkage</code>: whether to impose a hierarchical prior on the values of <code>delta</code> (cell type specific increase in expression of marker genes)</li>
</ul>
</div>
<div id="constructing-a-marker-gene-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#constructing-a-marker-gene-matrix" class="anchor"></a>Constructing a marker gene matrix</h2>
<p>Here we demonstrate a method of constructing the binary marker gene matrix that encodes our <em>a priori</em> knowledge of cell types.</p>
<p>For two types of cells (<code>Group1</code> and <code>Group2</code>) we know <em>a priori</em> several good marker genes, e.g.:</p>
<table class="table">
<thead><tr class="header">
<th>Cell type</th>
<th>Genes</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Group1</td>
<td>Gene186, Gene269, Gene526, Gene536, Gene994</td>
</tr>
<tr class="even">
<td>Group2</td>
<td>Gene205, Gene575, Gene754, Gene773, Gene949</td>
</tr>
</tbody>
</table>
<p>To use this in <code>cellassign</code>, we must turn this into a <em>named list</em>, where the names are the cell types and the entries are marker genes (not necessarily mutually exclusive) for each cell type:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">marker_gene_list &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">  <span class="dt">Group1 =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Gene186"</span>, <span class="st">"Gene269"</span>, <span class="st">"Gene526"</span>, <span class="st">"Gene536"</span>, <span class="st">"Gene994"</span>),</a>
<a class="sourceLine" id="cb23-3" data-line-number="3">  <span class="dt">Group2 =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Gene205"</span>, <span class="st">"Gene575"</span>, <span class="st">"Gene754"</span>, <span class="st">"Gene773"</span>, <span class="st">"Gene949"</span>)</a>
<a class="sourceLine" id="cb23-4" data-line-number="4">)</a>
<a class="sourceLine" id="cb23-5" data-line-number="5"></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(marker_gene_list))</a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="co">#&gt; List of 2</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="co">#&gt;  $ Group1: chr [1:5] "Gene186" "Gene269" "Gene526" "Gene536" ...</span></a>
<a class="sourceLine" id="cb23-9" data-line-number="9"><span class="co">#&gt;  $ Group2: chr [1:5] "Gene205" "Gene575" "Gene754" "Gene773" ...</span></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="co">#&gt; NULL</span></a></code></pre></div>
<p>We can then directly provide this to <code>cellassign</code> or turn it into a binary marker gene matrix first using the <code>marker_list_to_mat</code> function:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/marker_list_to_mat.html">marker_list_to_mat</a></span>(marker_gene_list))</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="co">#&gt;         Group1 Group2 other</span></a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="co">#&gt; Gene186      1      0     0</span></a>
<a class="sourceLine" id="cb24-4" data-line-number="4"><span class="co">#&gt; Gene205      0      1     0</span></a>
<a class="sourceLine" id="cb24-5" data-line-number="5"><span class="co">#&gt; Gene269      1      0     0</span></a>
<a class="sourceLine" id="cb24-6" data-line-number="6"><span class="co">#&gt; Gene526      1      0     0</span></a>
<a class="sourceLine" id="cb24-7" data-line-number="7"><span class="co">#&gt; Gene536      1      0     0</span></a>
<a class="sourceLine" id="cb24-8" data-line-number="8"><span class="co">#&gt; Gene575      0      1     0</span></a>
<a class="sourceLine" id="cb24-9" data-line-number="9"><span class="co">#&gt; Gene754      0      1     0</span></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co">#&gt; Gene773      0      1     0</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11"><span class="co">#&gt; Gene949      0      1     0</span></a>
<a class="sourceLine" id="cb24-12" data-line-number="12"><span class="co">#&gt; Gene994      1      0     0</span></a></code></pre></div>
<p>This has automatically included an <code>other</code> group for cells that do not fall into either type - this can be excluded by setting <code>include_other = FALSE</code>.</p>
</div>
<div id="adding-covariates" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-covariates" class="anchor"></a>Adding covariates</h2>
<p>Covariates corresponding to batch, sample, or patient-specific effects can be included in the <code>cellassign</code> model. For example, if we have two covariates <code>x1</code> and <code>x2</code>:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">N &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span>(example_sce)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2">x1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(N)</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">x2 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span>(N)</a></code></pre></div>
<p>We can construct a design matrix using the <code>model.matrix</code> function in R:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">X &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(<span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>x1 <span class="op">+</span><span class="st"> </span>x2)</a></code></pre></div>
<p>Note we explicitly set no intercept by passing in <code>0</code> in the beginning. We can then perform an equivalent cell assignment passing this in also:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">fit &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/cellassign/man/cellassign.html">cellassign</a></span>(<span class="dt">exprs_obj =</span> example_sce, </a>
<a class="sourceLine" id="cb27-2" data-line-number="2">                  <span class="dt">marker_gene_info =</span> example_marker_mat, </a>
<a class="sourceLine" id="cb27-3" data-line-number="3">                  <span class="dt">X =</span> X,</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">                  <span class="dt">s =</span> s, </a>
<a class="sourceLine" id="cb27-5" data-line-number="5">                  <span class="dt">learning_rate =</span> <span class="fl">1e-2</span>, </a>
<a class="sourceLine" id="cb27-6" data-line-number="6">                  <span class="dt">shrinkage =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb27-7" data-line-number="7">                  <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
</div>
</div>
<div id="technical" class="section level1">
<h1 class="hasAnchor">
<a href="#technical" class="anchor"></a>Technical</h1>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="co">#&gt; R version 3.6.0 (2019-04-26)</span></a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4"><span class="co">#&gt; Running under: macOS  10.15.2</span></a>
<a class="sourceLine" id="cb28-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb28-7" data-line-number="7"><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span></a>
<a class="sourceLine" id="cb28-8" data-line-number="8"><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></a>
<a class="sourceLine" id="cb28-9" data-line-number="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-10" data-line-number="10"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb28-11" data-line-number="11"><span class="co">#&gt; [1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8</span></a>
<a class="sourceLine" id="cb28-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb28-14" data-line-number="14"><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb28-15" data-line-number="15"><span class="co">#&gt; [8] methods   base     </span></a>
<a class="sourceLine" id="cb28-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-17" data-line-number="17"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb28-18" data-line-number="18"><span class="co">#&gt;  [1] cellassign_0.99.16          SingleCellExperiment_1.6.0 </span></a>
<a class="sourceLine" id="cb28-19" data-line-number="19"><span class="co">#&gt;  [3] SummarizedExperiment_1.14.1 DelayedArray_0.10.0        </span></a>
<a class="sourceLine" id="cb28-20" data-line-number="20"><span class="co">#&gt;  [5] BiocParallel_1.18.1         matrixStats_0.55.0         </span></a>
<a class="sourceLine" id="cb28-21" data-line-number="21"><span class="co">#&gt;  [7] Biobase_2.44.0              GenomicRanges_1.36.1       </span></a>
<a class="sourceLine" id="cb28-22" data-line-number="22"><span class="co">#&gt;  [9] GenomeInfoDb_1.20.0         IRanges_2.18.3             </span></a>
<a class="sourceLine" id="cb28-23" data-line-number="23"><span class="co">#&gt; [11] S4Vectors_0.22.1            BiocGenerics_0.30.0        </span></a>
<a class="sourceLine" id="cb28-24" data-line-number="24"><span class="co">#&gt; [13] BiocStyle_2.12.0           </span></a>
<a class="sourceLine" id="cb28-25" data-line-number="25"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb28-26" data-line-number="26"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb28-27" data-line-number="27"><span class="co">#&gt;  [1] reticulate_1.13        xfun_0.12              lattice_0.20-38       </span></a>
<a class="sourceLine" id="cb28-28" data-line-number="28"><span class="co">#&gt;  [4] colorspace_1.4-1       htmltools_0.4.0        yaml_2.2.0            </span></a>
<a class="sourceLine" id="cb28-29" data-line-number="29"><span class="co">#&gt;  [7] base64enc_0.1-3        rlang_0.4.2            pkgdown_1.4.1         </span></a>
<a class="sourceLine" id="cb28-30" data-line-number="30"><span class="co">#&gt; [10] RColorBrewer_1.1-2     lifecycle_0.1.0        GenomeInfoDbData_1.2.1</span></a>
<a class="sourceLine" id="cb28-31" data-line-number="31"><span class="co">#&gt; [13] tensorflow_2.0.0       stringr_1.4.0          zlibbioc_1.30.0       </span></a>
<a class="sourceLine" id="cb28-32" data-line-number="32"><span class="co">#&gt; [16] munsell_0.5.0          gtable_0.3.0           memoise_1.1.0         </span></a>
<a class="sourceLine" id="cb28-33" data-line-number="33"><span class="co">#&gt; [19] evaluate_0.14          knitr_1.27             tfruns_1.4            </span></a>
<a class="sourceLine" id="cb28-34" data-line-number="34"><span class="co">#&gt; [22] Rcpp_1.0.3             scales_1.1.0           backports_1.1.5       </span></a>
<a class="sourceLine" id="cb28-35" data-line-number="35"><span class="co">#&gt; [25] BiocManager_1.30.10    desc_1.2.0             jsonlite_1.6          </span></a>
<a class="sourceLine" id="cb28-36" data-line-number="36"><span class="co">#&gt; [28] XVector_0.24.0         fs_1.3.1               png_0.1-7             </span></a>
<a class="sourceLine" id="cb28-37" data-line-number="37"><span class="co">#&gt; [31] digest_0.6.23          stringi_1.4.5          bookdown_0.17         </span></a>
<a class="sourceLine" id="cb28-38" data-line-number="38"><span class="co">#&gt; [34] grid_3.6.0             rprojroot_1.3-2        tools_3.6.0           </span></a>
<a class="sourceLine" id="cb28-39" data-line-number="39"><span class="co">#&gt; [37] bitops_1.0-6           magrittr_1.5           RCurl_1.95-4.13       </span></a>
<a class="sourceLine" id="cb28-40" data-line-number="40"><span class="co">#&gt; [40] crayon_1.3.4           whisker_0.4            MASS_7.3-51.5         </span></a>
<a class="sourceLine" id="cb28-41" data-line-number="41"><span class="co">#&gt; [43] pheatmap_1.0.12        Matrix_1.2-18          assertthat_0.2.1      </span></a>
<a class="sourceLine" id="cb28-42" data-line-number="42"><span class="co">#&gt; [46] rmarkdown_2.0          rstudioapi_0.10        R6_2.4.1              </span></a>
<a class="sourceLine" id="cb28-43" data-line-number="43"><span class="co">#&gt; [49] compiler_3.6.0</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#basic-usage">Basic usage</a></li>
      <li><a href="#example-set-of-markers-for-tumour-microenvironment">Example set of markers for tumour microenvironment</a></li>
      <li>
<a href="#advanced-usage">Advanced usage</a><ul class="nav nav-pills nav-stacked">
<li><a href="#options-for-a-cellassign-call">Options for a <code>cellassign()</code> call</a></li>
      <li><a href="#constructing-a-marker-gene-matrix">Constructing a marker gene matrix</a></li>
      <li><a href="#adding-covariates">Adding covariates</a></li>
      </ul>
</li>
      <li><a href="#technical">Technical</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Allen Zhang, Kieran Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
