<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Annotate cells to cell types using cellassign — cellassign • cellassign</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Annotate cells to cell types using cellassign — cellassign" />
<meta property="og:description" content="Automatically annotate cells to known types based
on the expression patterns of
a priori known marker genes." />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cellassign</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/constructing-markers-from-purified-data.html">Constructing marker genes from purified scRNA-seq data</a>
    </li>
    <li>
      <a href="../articles/introduction-to-cellassign.html">Assigning single-cells to known cell types with CellAssign</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Annotate cells to cell types using cellassign</h1>
    
    <div class="hidden name"><code>cellassign.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Automatically annotate cells to known types based
on the expression patterns of
a priori known marker genes.</p>
    </div>

    <pre class="usage"><span class='fu'>cellassign</span>(<span class='no'>exprs_obj</span>, <span class='no'>marker_gene_info</span>, <span class='kw'>s</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>min_delta</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>X</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>B</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>shrinkage</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>n_batches</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>dirichlet_concentration</span> <span class='kw'>=</span> <span class='fl'>0.01</span>, <span class='kw'>rel_tol_adam</span> <span class='kw'>=</span> <span class='fl'>1e-04</span>,
  <span class='kw'>rel_tol_em</span> <span class='kw'>=</span> <span class='fl'>1e-04</span>, <span class='kw'>max_iter_adam</span> <span class='kw'>=</span> <span class='fl'>1e+05</span>, <span class='kw'>max_iter_em</span> <span class='kw'>=</span> <span class='fl'>20</span>,
  <span class='kw'>learning_rate</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>sce_assay</span> <span class='kw'>=</span> <span class='st'>"counts"</span>,
  <span class='kw'>return_SCE</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>num_runs</span> <span class='kw'>=</span> <span class='fl'>1</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>exprs_obj</th>
      <td><p>Either a matrix representing gene
expression counts or a <code>SummarizedExperiment</code>.
See details.</p></td>
    </tr>
    <tr>
      <th>marker_gene_info</th>
      <td><p>Information relating marker genes to cell types.
See details.</p></td>
    </tr>
    <tr>
      <th>s</th>
      <td><p>Numeric vector of cell size factors</p></td>
    </tr>
    <tr>
      <th>min_delta</th>
      <td><p>The minimum log fold change a marker gene must
be over-expressed by in its cell type</p></td>
    </tr>
    <tr>
      <th>X</th>
      <td><p>Numeric matrix of external covariates. See details.</p></td>
    </tr>
    <tr>
      <th>B</th>
      <td><p>Number of bases to use for RBF dispersion function</p></td>
    </tr>
    <tr>
      <th>shrinkage</th>
      <td><p>Logical - should the delta parameters
have hierarchical shrinkage?</p></td>
    </tr>
    <tr>
      <th>n_batches</th>
      <td><p>Number of data subsample batches to use in inference</p></td>
    </tr>
    <tr>
      <th>dirichlet_concentration</th>
      <td><p>Dirichlet concentration parameter for cell
type abundances</p></td>
    </tr>
    <tr>
      <th>rel_tol_adam</th>
      <td><p>The change in Q function value (in pct) below which
each optimization round is considered converged</p></td>
    </tr>
    <tr>
      <th>rel_tol_em</th>
      <td><p>The change in log marginal likelihood value (in pct)
below which the EM algorithm is considered converged</p></td>
    </tr>
    <tr>
      <th>max_iter_adam</th>
      <td><p>Maximum number of ADAM iterations
to perform in each M-step</p></td>
    </tr>
    <tr>
      <th>max_iter_em</th>
      <td><p>Maximum number of EM iterations to perform</p></td>
    </tr>
    <tr>
      <th>learning_rate</th>
      <td><p>Learning rate of ADAM optimization</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical - should running info be printed?</p></td>
    </tr>
    <tr>
      <th>sce_assay</th>
      <td><p>The <code>assay</code> from the input#' <code>SingleCellExperiment</code> to use: this assay
should always represent raw counts.</p></td>
    </tr>
    <tr>
      <th>return_SCE</th>
      <td><p>Logical - should a SingleCellExperiment be returned
with the cell
type annotations added? See details.</p></td>
    </tr>
    <tr>
      <th>num_runs</th>
      <td><p>Number of EM optimizations to perform (the one with the maximum
log-marginal likelihood value will be used as the final).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class <code>cellassign</code>. See <code>details</code></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><strong>Input format</strong>
<code>exprs_obj</code> should be either a
<code>SummarizedExperiment</code> (we recommend the
<code>SingleCellExperiment</code> package) or a
cell (row) by gene (column) matrix of
<em>raw</em> RNA-seq counts (do <strong>not</strong>
log-transform or otherwise normalize).</p>
<p><code>marker_gene_info</code> should either be</p><ul>
<li><p>A gene by cell type binary matrix, where a 1 indicates that a gene is a
marker for a cell type, and 0 otherwise</p></li>
<li><p>A list with names corresponding to cell types, where each entry is a
vector of marker gene names. These are converted to the above matrix using
the <code>marker_list_to_mat</code> function.</p></li>
</ul>

<p><strong>Cell size factors</strong>
If the cell size factors <code>s</code> are
not provided they are computed using the
<code>computeSumFactors</code> function from
the <code>scran</code> package.</p>
<p><strong>Covariates</strong>
If <code>X</code> is not <code>NULL</code> then it should be
an <code>N</code> by <code>P</code> matrix
of covariates for <code>N</code> cells and <code>P</code> covariates.
Such a matrix would typically
be returned by a call to <code>model.matrix</code>
<strong>with no intercept</strong>. It is also highly
recommended that any numerical (ie non-factor or one-hot-encoded)
covariates be standardized
to have mean 0 and standard deviation 1.</p>
<p><strong>cellassign</strong>
A call to <code>cellassign</code> returns an object
of class <code>cellassign</code>. To access the
MLE estimates of cell types, call <code>fit$cell_type</code>.
To access all MLE parameter
estimates, call <code>fit$mle_params</code>.</p>
<p><strong>Returning a SingleCellExperiment</strong></p>
<p>If <code>return_SCE</code> is true, a call to <code>cellassign</code> will return
the input SingleCellExperiment, with the following added:</p><ul>
<li><p>A column <code>cellassign_celltype</code> to <code>colData(sce)</code> with the MAP
estimate of the cell type</p></li>
<li><p>A slot <code>sce@metadata$cellassign</code> containing the cellassign fit.
Note that a <code>SingleCellExperiment</code> must be provided as <code>exprs_obj</code>
for this option to be valid.</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>example_sce</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>example_marker_mat</span>)

<span class='no'>fit</span> <span class='kw'>&lt;-</span> <span class='no'>em_result</span> <span class='kw'>&lt;-</span> <span class='fu'>cellassign</span>(<span class='no'>example_sce</span>[<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>example_marker_mat</span>),],
<span class='kw'>marker_gene_info</span> <span class='kw'>=</span> <span class='no'>example_marker_mat</span>,
<span class='kw'>s</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colSums.html'>colSums</a></span>(<span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span>(<span class='no'>example_sce</span>, <span class='st'>"counts"</span>)),
<span class='kw'>learning_rate</span> <span class='kw'>=</span> <span class='fl'>1e-2</span>,
<span class='kw'>shrinkage</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
<span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: SingleCellExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘SummarizedExperiment’ was built under R version 3.6.1</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘GenomicRanges’ was built under R version 3.6.1</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘S4Vectors’ was built under R version 3.6.1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘IRanges’ was built under R version 3.6.1</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘matrixStats’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘BiocParallel’ was built under R version 3.6.1</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘DelayedArray’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    aperm, apply, rowsum</span></div><div class='input'>

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Allen Zhang, Kieran Campbell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


